tests/core/test_signal.py

```diff
@@ -281,3 +282,25 @@ async def test_subscription_logs(caplog):
     assert "Making subscription" in caplog.text
     cache.close()
     assert "Closing subscription on source" in caplog.text
+
+
+async def test_signal_lazily_connects(RE):
+    mock_signal_rw = soft_signal_rw(int, "pva://mock_signal", name="mock_signal")
+    await mock_signal_rw.connect(mock=True)
+    RE(ensure_connected(mock_signal_rw, mock=True))
+    assert (
+        mock_signal_rw._connect_task
+        and mock_signal_rw._connect_task.done()
+        and not mock_signal_rw._connect_task.exception()
+    )
+
```

Comment:
It'd be good to see a test where the connect task fails, and then succeeds.

Reply:
could you elaborate please? what would be the reason of connection failure in the test?

Reply:
Because with your new connect logic,
``` python
can_use_previous_connection: bool = (
            self._connect_task
            and not (self._connect_task.done() and self._connect_task.exception())
)
```
a signal will reconnect if the previous connect
* doesn't exist (vacuously well tested)
* is `.done()` (we need a test for this)
* is `.exception()` (we need a test for this)
* uses `force_reconnect` (we need a test for this)